src/modules/verifiable-credential/services/vc-match.service.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  CredentialSubject,
+  VerifiableCredential,
+} from '@ew-did-registry/credentials-interface';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Checked, PEX } from '@sphereon/pex';
+import { Logger } from '../../logger/logger.service';
+import { PresentationDefinitionDto } from '../dtos/presentation-definition.dto';
+import { VCStorage } from '../entities/vc-storage.entity';
+import { VCStorageService } from './vc-storage.service';
+
+@Injectable()
+export class VCMatchService {
+  constructor(
+    private readonly logger: Logger,
+    @InjectRepository(VCStorage)
+    private readonly vcStorageService: VCStorageService
+  ) {
+    this.logger.setContext(VCMatchService.name);
+  }
+
+  private checkedToString(checked: Checked[]): string {
+    return checked
+      .filter((err) => err.message)
+      .map((err) => err.message)
+      .join(', ');
+  }
+
+  async matchCredentials(
+    presentationDefinition: PresentationDefinitionDto,
+    holder: string
+  ): Promise<VerifiableCredential<CredentialSubject>[]> {
```

Comment:
I think that instead of returning the `Credentials` we should just return the `SelectResults`. This is because the client will need more than just the credentials to display to the user. For instance, I think `SelectResult.matches` is required so that the UI can guide the client as to which credentials can meet which submission requirement